src/client.rs

```diff
@@ -52,12 +52,17 @@ macro_rules! impl_inner_call {
                     return res;
                 },
                 Err(e) => {
-                    warn!("call retry:{}/{} {:?}", errors.len() + 1 , $self.config.retry(), e);
-                    errors.push(e);
-                    if errors.len() as u8 == $self.config.retry() {
+                    let failed_attempts = (errors.len() + 1) as u8;
+
+                    if failed_attempts > $self.config.retry() {
```

Comment:
I think in the rather subtle case of `retry = 255` the if condition will never satisfy (or panic in debug mode)

It may be better to let `failed_attempts` usize and cast retry instead

Reply:
Using `usize` for the `retry` value feels wrong, the value of `retry` is independent from the platforms pointer width. I'll think of something.

Reply:
See 3d08748.